test/common.js

```diff
@@ -0,0 +1,19 @@
+var yaml = require('js-yaml');
+
+module.exports.getDiag = function (body) {
+    var yamlStart = body.indexOf('  ---');
+    var yamlEnd = body.indexOf('  ...\n');
+    var diag = body.slice(yamlStart, yamlEnd).split('\n').map(function (line) {
+        return line.slice(2);
+    }).join('\n');
+
+    // The stack trace will vary depending on where the code is run, so just
+    // strip it out.
+    var withStack = yaml.safeLoad(diag);
+    delete withStack.stack;
+    return withStack;
+}
+
+module.exports.stripFullStack = function (output) {
+    return output.replace(/^\W+at.*:\d+:\d+.*$\W/gm, '');
```

Comment:
We could strip out all frames that don't involve the `test/` directory. So the result would look like this:

```
           Error: should be equivalent                                         
              [... frames omitted ...]
              at Test.<anonymous> (/Users/fmurphy/src/tape/test/undef.js:36:11)
              [... frames omitted ...]
```

Would that be a good compromise?